ci/spelling.sh

```diff
@@ -1,7 +1,12 @@
 #!/usr/bin/env bash
 
+set -eo pipefail
+
 function spellcheck () {
-    git diff --name-only --diff-filter=AM main $HEAD | grep '\.md$' | xargs --no-run-if-empty -L1 npx cspell --show-suggestions -c ./ci/spelling-config.json
+    FAILURE=0
```

Comment:
Can we leave the markdown filtering as is ? the new filter does not seem to work when I tested it locally

```
personalmacbook:tag-security demo$ git diff --name-only --diff-filter=AM main $HEAD -- ./**/*.md ./*.md 
personalmacbook:tag-security demo$ git diff --name-only --diff-filter=AM main $HEAD
ci/links.sh
ci/lint.sh
ci/setup.sh
ci/spelling.sh
security-whitepaper/v2/cloud-native-security-whitepaper-it.md
security-whitepaper/v2/cloud-native-security-whitepaper-simplified-chinese.md
personalmacbook:tag-security demo$ git diff --name-only --diff-filter=AM main | grep '\.md$'
security-whitepaper/v2/cloud-native-security-whitepaper-it.md
security-whitepaper/v2/cloud-native-security-whitepaper-simplified-chinese.md
```

Reply:
I also tested this with `make spelling` and new filtering did not pick the markdown files changed but the older filtering did.

Reply:
This is coming from the missing `shopt -s globstar` on your local shell (its responsible for expanding `**`) , which I have enabled on mine. I added it to the scripts, now it works as expected. Sorry about that.